src/Repository/AdherentRepository.php

```diff
@@ -499,6 +499,28 @@ public function countByGender(): array
         return $this->createQueryBuilder('a', 'a.gender')
             ->select('a.gender, COUNT(a.id) AS count')
             ->where('a.adherent = 1')
+            ->andWhere('a.status = :status')
+            ->setParameter('status', Adherent::ENABLED)
+            ->groupBy('a.gender')
+            ->getQuery()
+            ->getArrayResult()
+        ;
+    }
+
+    public function countByGenderManagedBy(Adherent $referent): array
+    {
+        if (!$referent->isReferent()) {
+            return [];
```

Comment:
Perhaps it would be better to throw an exception as it looks like a dev mistake to call this method with non referent.